### DISCUSSION TITLE
How to get an `if` statement to work?

### DISCUSSION THREAD

**Discussion by midas0441** [SilentVoid13/Templater#285] [Help] [answered]
---
created: 2021-07-17 Sat 17:16:03
tag: 
---

I am having a bit of trouble with a template that contains a rather simple `if` statement. The template is called `tester` with the code below:

```
<%*
  let title = tp.file.title;
  let tags = "replacethis";
  if (title.startsWith("tryme")) {
				let title = "tryme " + tp.date.now("YYYY-MM-DD");
				let tags = "tryme"; 
				}
	tR += tags;
%>
---
created: <% tp.date.now("YYYY-MM-DD") %>
---
<% tp.file.cursor() %>
```

In an empty note named `tryme`, I invoked the above template. The output is:
```
replacethis
---
created: 2021-07-17
---
```

The Expected Result was:

```
tryme
---
created: 2021-07-17
---
```

If I move `tR+=tags;` inside the `if` statement, it produces the Expected Result. So, the code is obviously getting inside the `if` statement, as expected. But, why doesn't the original code produce the Expected Result?

**Comment by megclaypool** [comment, marked as answer]
On lines 5 and 6 you have:
```js
let title = "tryme " + tp.date.now("YYYY-MM-DD");
let tags = "tryme"; 
```
I think by using the let keyword here, you are redeclaring the variables. `let` variables are scope limited, so those values only apply inside that function block. Try getting rid of the `let` on both those lines, and see if your function works as expected 😀

**Reply by midas0441** [reply]
thank you, @megclaypool! It would've never occurred to me to look into the `let` statement. I am learning on the fly and am grateful for your tip.
